### PR TITLE
Upgrade to vertx 4 and other dependencies

### PR DESCRIPTION
**Pull request by siladu**
Upgrade to vertx 4.2.3 to match besu and signers
Upgrade signers to 2.0.0
Upgrade besu metrics to 21.10.9
Upgrade various deps and plugins to match signers and ethsigner

[owasp dependency-check](https://owasp.org/www-project-dependency-check/) report is passing locally.

https://github.com/ConsenSys/web3signer/issues/484

### CODE REVIEW

**Review by lucassaldanha — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## Next Version
+### Breaking Changes
+- Update Vertx to 4.x.
+- Update signers to 2.0.0
+- Update various other dependencies to latest versions.
```
[the remark this answers is not in the document]
Comment (siladu) [reply CHANGELOG.md:7]: good catch

**Review by usmansaleem — APPROVED** [review]
LGTM

File: gradle/versions.gradle [hunk 2]
```diff
@@ -13,25 +13,25 @@
...
-    dependency 'commons-lang:commons-lang:2.6'
+    dependency 'commons-cli:commons-cli:1.5.0'
+    dependency 'commons-io:commons-io:2.11.0'
+    dependency 'org.apache.commons:commons-lang3:3.12.0'
 
-    dependency 'info.picocli:picocli:4.5.1'
+    dependency 'info.picocli:picocli:4.6.2'
 
-    dependencySet(group: 'io.vertx', version: '3.9.10') {
+    dependencySet(group: 'io.vertx', version: '4.2.3') {
```
Comment (jframe) [inline gradle/versions.gradle:34]: Latest vertx is now 4.2.4

Comment (siladu) [reply gradle/versions.gradle:34]: Matching signer version at 4.2.3

**Pull request merged**